src/data_race.rs

```diff
@@ -486,7 +476,33 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriEvalContextExt<'mir, 'tcx> {
         atomic: AtomicReadOp,
     ) -> InterpResult<'tcx, ScalarMaybeUninit<Tag>> {
         let this = self.eval_context_ref();
+        // This will read from the last store in the modification order of this location. In case
+        // weak memory emulation is enabled, this may not be the store we've picked to actually read from and return.
```

Comment:
I was confused for a bit here by this "store we've picked" -- we haven't picked anything yet, right? This is about the store that we *will* pick a few lines down in `buffered_read`, isn't it?